MigrationTutorial/Blog-Post.md

```diff
@@ -0,0 +1,658 @@
+        var consumableToDelete = new List<Consumable>();
+
+        migration.RenameProperty(nameof(Consumable), "Price", nameof(Consumable.LastPurchasedPrice));
+
+        for (var i = 0; i < newConsumables.Count(); i++)
+        {
+            var currConsumable = newConsumables.ElementAt(i);
+
+            // remove duplicates since ProductId is the new PrimaryKey
+            if (distinctConsumableId.Contains(currConsumable.ProductId))
+            {
+                consumableToDelete.Add(currConsumable);
+            }
+            else
+            {
+                distinctConsumableId.Add(currConsumable.ProductId);
+            }
+        }
+
+        consumableToDelete.ForEach(x => migration.NewRealm.Remove(x));
+    }
+
+    ConvertConsumableToTool(migration, oldSchemaVersion, "GlueHolder"); // 22)
+
+    ConvertConsumableToTool(migration, oldSchemaVersion, "Brush");  // 23)
+}
+
+private static void ConvertConsumableToTool(Migration migration, ulong oldSchemaVersion, string consumableType)
+{
+    // it's assumed that there's always 1 and 1 only type of Consumable
+    var oldConsumable = ((IQueryable<RealmObject>)migration.OldRealm.DynamicApi.All("Consumable")).Filter("_Type == $0", consumableType).FirstOrDefault();  // 24)
+    if (oldConsumable == null)
+    {
+        Logger.LogWarning($"No consumable was found with type {consumableType}. Nothing to convert.");
+        return;
+    }
+
+    Supplier consumableSupplier = null;
+    string consumableBrand = string.Empty;
+
+    // no suppliers if coming from schemaVersion 1
+    if (oldSchemaVersion > 1)   // 25)
+    {
+        var supplierId = oldConsumable.DynamicApi.Get<RealmObject>("Supplier").DynamicApi.Get<ObjectId>("Id");
+        consumableSupplier = migration.NewRealm.All<Supplier>().Filter("Id == $0", supplierId).FirstOrDefault();
+        consumableBrand = oldConsumable.DynamicApi.Get<string>("Brand");
+    }
+
+    migration.NewRealm.Add(new MachineryAndTool()   // 26)
+    {
+        Type = Type.ManufacturingTool,
+        Status = OperationalStatus.Functioning,
+        AssignedMaintainer = null,
+        ToolName = oldConsumable.DynamicApi.Get<string>("_Type").ToString(),
+        Supplier = consumableSupplier,
+        Brand = consumableBrand
+    });
+
+    var newConsumables = migration.NewRealm.All<Consumable>();
+
+    // ProductId could be empty because of human error
+    var consumableProductId = oldConsumable.DynamicApi.Get<string>("ProductId");
+    if (consumableProductId != string.Empty)
+    {
+        var consumableToRemove = newConsumables.Where(x => x.ProductId == consumableProductId).First();
+        migration.NewRealm.Remove(consumableToRemove);  //27)
+    }
+}
+```
+
+---
+
+<br/>
+
+## Conclusion
+Schema migrations may sometimes be confusing. This blog-post wanted to help shedding more light on how to perform some common migrations in Realm. More specifically, it was shown how you could:
```

Comment:
I think that this list is extremely valuable, because it summarises the common cases you wanted to show, while maybe they get lost in the migrations code. 

I am not sure exactly how, but I think it would be useful to underline those in code or the blog text when we do those, so that if someone is just giving a look at the blog post can find those at a glance.